### DISCUSSION TITLE
What is push time?

### DISCUSSION THREAD

**Discussion by hwlr0** [SmileiPIC/Smilei#771] [Q&A] [answered]
Hello!

My question is pretty much all in the title. I am trying to optimize the parameters of my code and its environment for more efficient use of my computer time, and I am wondering what is the "push time" column in Smilei's printed output.

From what I observed, it seems that the higher its value, the less efficiently the computer is used. It also seems to correlate quite well (not perfectly tought) with the output of the diagnostics. But really these are just guesses, so a real explanation would be much appreciated!

Cheers!
Howel

**Comment by mccoys** [comment, marked as answer]
Hi

The push time is simply the total time to do one iteration, divided by the number of particles, and multiplied by the number of cores detected (which sometimes is not really the actual number of cores, depending on how the system provides this information). It represents somewhat loosely the time to compute one timestep for one particle.

**Reply by hwlr0** [reply]
Ok thanks!
